ludwig/datasets/mixins/download.py

```diff
@@ -46,6 +46,41 @@ def download_raw_dataset(self):
     def download_urls(self):
         return self.config["download_urls"]
 
+
+class GZipDownloadMixin:
+    """Downloads the gzip archive file containing the training data and extracts the contents."""
+
+    config: dict
+    raw_dataset_path: str
+    raw_temp_path: str
+
+    def download_raw_dataset(self):
+        """
+        Download the raw dataset and extract the contents of the zip file and
+        store that in the cache location.
+        """
+        """
+                Download the raw dataset and contents of the gzip file
+                onto the _raw directory.
+                """
+        os.makedirs(self.raw_temp_path, exist_ok=True)
+        for file_download_url in self.download_urls:
+            filename = file_download_url.split('/')[-1]
+            urllib.request.urlretrieve(file_download_url, os.path.join(self.raw_temp_path, filename))
+            size_of_original_file = len(filename)
+            gzip_content_file = filename[:size_of_original_file - 3]
```

Comment:
It's possible that the extension might be `.gzip` instead of `.gz`, right?  To be safe, we could split filename on `.` and then join all but the last:

```
gzip_content_file = '.'.join(filename.split('.')[:-1])
```

Reply:
Fixed